frontend/src/components/CardBack.tsx

```diff
@@ -57,11 +57,33 @@ const MeaningContent = ({
   const [showEmojis, setShowEmojis] = useState(false);
   const [showIndicators, setShowIndicators] = useState(false);
 
+  const { theme, colors } = useTheme();
+  const headerColor =
+    theme === 'light'
+      ? colors.primary || '#5ba20f'
+      : colors.primary || '#AEEA00';
+
   return (
     <View style={styles.meaningContentContainer}>
       <View style={styles.titleSection}>
-        <Ionicons name="bulb-outline" size={22} color={PRIMARY_ACCENT_COLOR} />
-        <Text style={[styles.stepTitle, { color: PRIMARY_ACCENT_COLOR }]}>
+        <Ionicons name="bulb-outline" size={22} color={headerColor} />
+        <Text
+          style={[
+            styles.stepTitle,
+            {
+              color: headerColor,
+              textShadowColor:
+                theme === 'light'
+                  ? 'transparent'
+                  : (colors.background ?? '#000') + '66',
+              textShadowOffset:
+                theme === 'light'
+                  ? { width: 0, height: 0 }
+                  : { width: 0, height: 1 },
+              textShadowRadius: theme === 'light' ? 2 : 2,
```

Comment:
All of these should be part of the useTheme hook. We should ideally not need to do these if-checks outside of the hook